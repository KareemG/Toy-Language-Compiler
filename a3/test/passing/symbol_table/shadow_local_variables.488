
{
    var a : integer
    var b : integer
    
    function foo : integer
    {
        var a : integer
        var b : integer
        
        a := 20
        b := 30
        return with a + b
    }
    
    a := 2
    b := foo
}