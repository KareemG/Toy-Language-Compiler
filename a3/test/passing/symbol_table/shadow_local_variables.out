{
 Symbols:
 b: integer
 a: integer
 foo: integer
 foo : integer{
 Symbols:
 b: integer
 a: integer
 }
